### PR TITLE
[DEITS] Add progress and telemetry events to import/export

### PR DESCRIPTION
**Pull request by innerdvations**
### What does it do?

- transfer engine progress stream now emits transfer events (start, finish, error)
- prefix stage events with "stage::"
- rename "complete" event -> "finish"
- adds telemetry events to import/export commands
- adds missing tests for progress stream 

### Why is it needed?

To improve the events available on the progress stream and add telemetry data for the CLI import/export command.

### How to test it?

Unit tests have been provided, but you can play around with events such as engine.progress.stream.on('stage::finish') and .on('transfer::start')

### CODE REVIEW

**Review by Bassel17 — APPROVED** [review]

File: packages/core/strapi/lib/commands/transfer/import.js [hunk 1]
```diff
@@ -82,6 +82,26 @@ module.exports = async (opts) => {
   try {
     logger.info('Starting import...');
 
+    const progress = engine.progress.stream;
+    const telemetryPayload = (/* payload */) => {
+      return {
```
Comment (Convly) [inline packages/core/strapi/lib/commands/transfer/import.js:87]: Shouldn't this be wrapped in an `eventProperties` object?

File: packages/core/data-transfer/lib/engine/index.ts [hunk 2]
```diff
@@ -343,9 +360,13 @@ class TransferEngine<
       await this.transferLinks();
       await this.transferConfiguration();
 
+      this.#emitTransferUpdate('finish');
```
Comment (Convly) [inline packages/core/data-transfer/lib/engine/index.ts:363]: If we consider the transfer started before the bootstrap & the init, shouldn't we wait for the close lifecycle to be run before sending the finish event?

File: packages/core/strapi/lib/commands/transfer/export.js [hunk 3]
```diff
@@ -77,6 +77,29 @@ module.exports = async (opts) => {
...
+    const telemetryPayload = (/* payload */) => {
+      return {
+        eventProperties: {
+          source: engine.sourceProvider.name,
+          destination: engine.destinationProvider.name,
+        },
+      };
+    };
+
+    progress.on('transfer::start', (payload = undefined) => {
```
Comment (Convly) [inline packages/core/strapi/lib/commands/transfer/export.js:91]: `= undefined` is the default's default value, no? Why do we need to specify it here? :thinking:

Comment (innerdvations) [reply packages/core/strapi/lib/commands/transfer/export.js]: The original reason was setting a default parameter thinking about typescript complaining in the future if you don't send it 😆

File: packages/core/data-transfer/lib/engine/index.ts (hunk 2 above)
Comment (innerdvations) [reply packages/core/data-transfer/lib/engine/index.ts]: Yeah, I was considering it a success if the actual data transfered, but I can see it either way. If we're still able to rollback in close, it should definitely be after, so I've moved it.

**Review by christiancp100 — APPROVED** [review]

**Review by Convly — APPROVED** [review]

**Pull request merged**